examples/features/gracefulstop/server/main.go

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Binary server demonstrates how to gracefully stop a gRPC server.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc"
+	pb "google.golang.org/grpc/examples/helloworld/helloworld"
+)
+
+var port = flag.Int("port", 50052, "port number")
+
+// server is used to implement helloworld.GreeterServer.
+type server struct {
+	pb.UnimplementedGreeterServer
+}
+
+// SayHello implements helloworld.GreeterServer.
+func (s *server) SayHello(_ context.Context, _ *pb.HelloRequest) (*pb.HelloReply, error) {
+	return &pb.HelloReply{Message: "Hello"}, nil
+}
+
+func main() {
+	flag.Parse()
+
+	address := fmt.Sprintf(":%v", *port)
+	lis, err := net.Listen("tcp", address)
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// Create a channel to listen for OS signals.
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+
+	s := grpc.NewServer()
+	pb.RegisterGreeterServer(s, &server{})
+
+	go func() {
+		// Wait for an OS signal for graceful shutdown.
+		<-stop
+		fmt.Println("Shutting down server...")
+		s.GracefulStop()
+		close(stop)
```

Comment:
Is this OK? Closing the channel given to `signal.Notify`?  What if ^C is pressed again after the channel is closed?  Wouldn't it panic?

Better to just do a `timer := time.AfterFunc(10*time.Second, s.Stop); defer timer.Stop(); s.GracefulStop()` here?

Reply:
Done